clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -613,7 +602,24 @@ private KafkaConsumer(ConsumerConfig config,
     }
 
     /**
-     * Incrementally subscribes to the given list of topics and uses the consumer's group management functionality
+     * Get the current subscription. Will return the same topics used in the most recent call to
+     * {@link #subscribe(List, ConsumerRebalanceListener)}, or an empty set if no such call has been made.
+     * @return The set of topics currently subscribed to
+     */
+    public Set<String> subscription() {
+        acquire();
+        try {
+            return Collections.unmodifiableSet(this.subscriptions.subscription());
```

Comment:
okay, this is subtle but worth noting, imho. The javadoc states that this method returns a snapshot of the current subscription, an immutable subscription, by the way. But by wrapping on ``Collections.unmodifiableSet``, this method is returning a *dynamic unmodifiable view* of the underlining Set (subscription). Therefore, when subscription change the Set view returned by this method will change too, so it's not a snapshot anymore. If the goal is to return an immutable *snapshot* then we should return a defensive copy as below:

```
return Collections.unmodifiableSet(new HashSet<>(this.subscriptions.subscription()));
```

Then, we are effectively returning a immutable snapshot of the subscriptions. Does it make sense?